### PR TITLE
Switch login to username password

### PR DESCRIPTION
**Pull request by begedin**
- [Asana task: Switch code login method to a username/password prompt](https://app.asana.com/0/26202368814744/41896304172551)
# Description

This PR removes the authorize command and instead adds a prompt to commands that require github authentication.
- Affected commands - "publish", "hotfix" and "ready".
- On successful authentication, the oauth key is stored on the machine, so there is no prompt a second time.
# ToDo
- [x] Write more specs
- [x] Mask the password input

### CODE REVIEW

**Comment by venkatd** [comment]
Looks good code is really clean

**Comment by venkatd** [comment]
(pun not intended)

File: bin/code [hunk 1]
```diff
@@ -52,6 +40,9 @@ desc 'Upload your feature branch to GitHub'
 command :publish, :pr do |c|
   c.switch [:master], desc: 'Make the pull request based off the master branch'
   c.action do |global_options, options, args|
+
+    $gitHubAPI.ensure_authorized
```
Comment (venkatd) [inline bin/code]: Can we make this fit the ruby conventions? Not sure if globals are usually defined with camel casing.

**Comment by venkatd** [comment]
Looks good but just a minor change

**Comment by begedin** [comment]
Renamed the variable. This is ready for merge.

**Pull request merged**